money.go

```diff
@@ -4,28 +4,59 @@ import (
 	"errors"
 )
 
-type amount struct {
-	val int
+// Amount is a datastructure that stores the amount being used for calculations
+type Amount struct {
+	val int64
 }
 
 // Money represents monetary value information, stores
 // currency and amount value
 type Money struct {
-	amount   *amount
+	amount   *Amount
 	currency *Currency
 }
 
 var calc *calculator
 
-// New creates and returns new instance of Money
-func New(Amount int, code string) *Money {
+func newAmount(amount interface{}) (*Amount, error) {
```

Comment:
Having multiple cases in a type switch is valid, but the `amt` value, in this scenario, is left as an `interface{}`. Using the switched value as *one of* the types in the case statement is not supported and is a build error.